test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/AllowListVerificationProviderTests.cs

```diff
@@ -401,323 +371,491 @@ public AllowListVerificationProviderTests(SigningTestFixture fixture)
 
                     // Assert
                     result.Valid.Should().BeTrue();
-                    totalWarningIssues.Count().Should().Be(1);
-                    totalWarningIssues.First().Code.Should().Be(NuGetLogCode.NU3034);
-                    totalWarningIssues.First().Message.Should().Be(_noClientAllowList);
+                    totalWarningIssues.Count().Should().Be(0);
                 }
             }
         }
 
         [CIOnlyFact]
-        public async Task GetTrustResultAsync_VerifyWithEmptyClientCertificateAllowList_WarnsWhenRequiredAndAllowUntrustedAsync()
+        public async Task GetTrustResultAsync_AuthorSignedPackage_WithEmptyRepositoryCertificateAllowList_RepositoryAndClientListNotRequired_SuccessAsync()
```

Comment:
I feel like all the combinations of null/empty/non-empty client/repository allow list and required/not required client/repository allow lists could be simplified with a lot less code duplication using `[Theory]` and `[MemberData(...)]`.  The matrix could even include verifier settings.